lib/broadway_rabbitmq/amqp_client.ex

```diff
@@ -265,6 +256,76 @@ defmodule BroadwayRabbitMQ.AmqpClient do
       {:error, :timeout}
   end
 
+  defp get_channel(%{connection: {:custom_pool, module, args}}) do
+    case module.checkout_channel(args) do
+      {:ok, channel} ->
+        Process.link(channel.pid)
+        {:ok, channel}
+
+      {:error, exception} when is_exception(exception) ->
```

Comment:
My bad, is_exception is too recent apparently. Use this

```suggestion
      {:error, %{__exception__: true}} = exception ->
```

Reply:
we need to change the checkout_channel too

Reply:
yep!